### DISCUSSION TITLE
Documentation re data storage

### DISCUSSION THREAD

**Discussion by aubreyz** [tadashi-aikawa/obsidian-various-complements-plugin#159] [Q&A]
Perhaps there needs to be some documentation about where data is stored. 

I was trying to synchronise this plugin across two different vaults. I want the same settings, but because the vaults have different purposes I don't want to copy across the 

- Intelligent suggestion prioritization
- Custom dictionaries

The dictionaries are easy - I know where they are.  But I cannot work out where data for Intelligent suggestion prioritization is stored...
It does not seem to be in data.json (which I would copy across with settings) - where is it?

**Comment by tadashi-aikawa** [comment]
> where is it?

@aubreyz 
`selectionHistoryTree` in `data.json`

```
{
  .
  .
  "selectionHistoryTree": {
    "Another Quick Switcher": {
      "🦉Another Quick Switcher": {
        "internalLink": {
          "count": 75,
          "lastUpdated": 1659405076322
        }
      }
    },
    "Various Complements": {
      "🦉Various Complements": {
        "internalLink": {
          "count": 100,
          "lastUpdated": 1659404627884
        }
      }
    },
    "Obsidian": {
      "Obsidian": {
        "internalLink": {
          "count": 169,
          "lastUpdated": 1659775170493
        }
      }
    },
.
.
}
```

**Comment by aubreyz** [comment]
Ah, yes this reminds me -- I could not find them because they had all been cleared! This was a vault I had not used for 30 days, so all the intelligence it had accumulated was eliminated. 

This is about the only thing I still find a bad design in this great plugin. 

Intelligent suggestion prioritization works great - much better than others ways to order suggestions which have been suggested (by length etc). However it makes absolutely no sense to me that they are cleared after 30 days. 

I can see that there is a reason to keep the data.json file size within reasonable limits -- but eliminating intelligence based on time makes no sense at all to me. I think that if you want to restrict the file size, you should just keep the most recently used number of entries (say 200), not eliminate by time and then everything is lost and you start again.

**Reply by tadashi-aikawa** [reply]
Please wait to implement https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/131#issuecomment-1126649056.

It depends on the person what they need. In fact, I do not need history deletion by size.
